### DISCUSSION TITLE
Calculating the ROI

### DISCUSSION THREAD

**Discussion by PatWekan** [nickpadd/EuropeanFootballLeaguePredictor#73] [General]
Hello

could you provide some backgrounds about calculating the ROI? Do you use the bets for which you detect a value or how is it calculated?

Thanks for

**Comment by nickpadd** [comment]
For the evaluation the process is as follows:

The model trains on the data of all the seasons except for one season that is left out for it to backtest on.
For every match in the season that is evaluated the model runs and produces it's estimated probabilities. 
It then compares it's probabilities with the odds provided by the betting company (by converting the odds to implied probabilities).

The percentage of the available bank that the model bets on is decided by a custom version of the [Kelly Criterion](https://en.wikipedia.org/wiki/Kelly_criterion). More specifically it uses the Quarter Kelly Criterion (Kelly Criterion divided by 4) but with a maximum cap in the percentage of the current bank it can bet (this cap can be adjusted [here](https://github.com/nickpadd/EuropeanFootballLeaguePredictor/blob/1afb02d20c2f4aab2387a23f540e6950a44b7256/europeanfootballleaguepredictor/config/config.yaml#L12C1-L12C69)). 

The model bets on the following seperately:
Home win, Draw, Away win, Over 2.5, Under 2.5

After betting on the whole season the ROI metric is produced:

$` ROI = 100*\frac{FinalBank - InitialInvestment}{InitialInvestment} `$

It is a metric that basically tells you what percentage of the Initial Investment is your net profit.
Finally I also produced some aggregate ROI metrics (as I explain in [here](https://github.com/nickpadd/EuropeanFootballLeaguePredictor/tree/master#running-the-scripts-for-the-upcoming-matches)) to evaluate weather betting on multiple betting categories could be profitable.

**Comment by PatWekan** [comment]
Hello Nick,

thanks for the background. So you use a customized Kelly criterion for betting the value bets you have detected.

I will jump deeper into it but many thanks for the backgrounds from your side and your implementation so far

Pat
